src/storage/cache/rpc/server.rs

```diff
@@ -0,0 +1,478 @@
+                    req_header.seq
+                );
+                let req_buffer: &mut BytesMut = unsafe { &mut *self.inner.req_buf.get() };
+                self.inner
+                    .dispatch_handler
+                    .dispatch(req_header, req_buffer, done_tx)
+                    .await;
+            }
+        } else {
+            debug!("Inner request type is not matched: {:?}", req_header.op);
+        }
+    }
+
+    /// Keep the connection and get the handler for the connection.
+    pub async fn run(&self) {
+        // Dispatch the handler for the connection
+        debug!("RpcServerConnection::run");
+
+        // TODO: copy done_tx to the worker pool
+        let (done_tx, mut done_rx) = mpsc::channel::<Vec<u8>>(10000);
+
+        // Send response to the stream from the worker pool
+        // Worker pool will handle the response sending
+        let inner_conn = Arc::clone(&self.inner);
+        tokio::spawn(async move {
+            debug!("Start to send response to the stream from client");
+            loop {
+                if let Some(resp_buffer) = done_rx.recv().await {
+                    debug!("Recv buffer from done_tx channel, try to send response to the stream");
+                    // Send response to the stream
+                    if let Ok(res) = inner_conn.send_response(&resp_buffer).await {
+                        debug!("Sent file block response successfully: {:?}", res);
+                    } else {
+                        error!("Failed to send file block response");
+                    }
+                } else {
+                    info!("done_rx channel is closed, stop sending response to the stream");
+                    break;
+                }
+            }
+        });
+
+        let start_time = std::time::Instant::now();
+        loop {
+            // Receive the request header
+            let req_header = match self.inner.recv_header().await {
+                Ok(header) => {
+                    debug!("Received request header: {:?}", header);
+                    header
+                }
+                Err(err) => {
+                    debug!("Failed to receive request header: {:?}", err);
+                    return;
+                }
+            };
+
+            if req_header.seq == 1000 {
+                let end_time = std::time::Instant::now();
+                info!("Total time: {:?}", end_time - start_time);
+            }
+            // Dispatch the handler for the connection
+            self.dispatch(req_header, done_tx.clone()).await;
+        }
+    }
+}
+
+/// The worker factory for the RPC connection.
+#[derive(Clone, Debug)]
+pub struct RpcConnWorkerFactory<T>
+where
+    T: RpcServerConnectionHandler + Send + Sync + 'static,
+{
+    /// Global worker pool for the RPC connection, shared by all connections.
+    worker_pool: Arc<WorkerPool>,
+    /// The handler for the connection
+    dispatch_handler: T,
```

Comment:
dispatch_handler is not use by the struct, there will be warnning?

Reply:
Yes, it is better to be removed.